tezos/context/src/working_tree/serializer.rs

```diff
@@ -497,6 +569,79 @@ fn deserialize_hash_id(data: &[u8]) -> Result<(Option<HashId>, usize), Deseriali
     }
 }
 
+fn deserialize_shaped_directory(
+    data: &[u8],
+    storage: &mut Storage,
+    store: &ContextKeyValueStore,
```

Comment:
`store` + `storage` is super confusing, we need to figure out a better name for `storage`.

Reply:
For `ContextKeyValueStore` we use `repo` or `repository` elsewhere, I think we should at least use that elsewhere to make things less ambiguous.

Reply:
And for the two cases where I see `store` being used directly now (instead of just passed around), I see it is always about obtaining or creating a new shape. I think it would be better if here we passed around a wrapper that is specifically about shapes, so it would be just `shapes` instead and also `shapes.get(id)` and `shapes.create(dir)`, what do you think?